### PR TITLE
fix: support custom storage endpoint suffix

### PR DESCRIPTION
**Pull request by cvvz**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_storageaccount.go [hunk 1]
```diff
@@ -160,6 +162,14 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 		subsID = accountOptions.SubscriptionID
 	}
 
+	if len(accountOptions.StorageEndpointSuffix) == 0 {
+		accountOptions.StorageEndpointSuffix = az.cloud.Environment.StorageEndpointSuffix
+	}
+
+	if accountOptions.CreatePrivateEndpoint {
+		privateDNSZoneName = fmt.Sprintf(privateDNSZoneName, accountOptions.StorageEndpointSuffix)
```
Comment (andyzhangx) [inline pkg/provider/azure_storageaccount.go:170]: don't use same var, otherwise `privateDNSZoneName` would be appened again and again

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Review by andyzhangx — COMMENTED** [review]
/hold

**Review by andyzhangx — COMMENTED** [review]
I think it's better generate `privateDNSZoneName` when it's needed, use a global env var may not be a good idea since file and blob use different privateDNSZoneName.

**Comment by cvvz** [comment]
> I think it's better generate `privateDNSZoneName` when it's needed, use a global env var may not be a good idea since file and blob use different privateDNSZoneName.

I have removed the global env var, I agree that we should avoid using global variables. `privateDNSZoneName` can be assembled at the beginning of `EnsureStorageAccout` then pass it as function parameter to avoid assembling it everytime when using it.

I will introduce `StorageType` in this [PR](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2992/files) to support blob csi driver.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/hold canel

**Pull request merged**